recipes-bsp/u-boot/files/0008-board-siemens-iot2050-Pass-DDR-size-from-FSBL.patch

```diff
@@ -0,0 +1,119 @@
+
+Also now for some variants, the DDR size may > 2GB, so borrow some code
+from the TI evm to iot2050 to support more than 2GB DDR.
+
+Signed-off-by: Baocheng Su <baocheng.su@siemens.com>
+---
+ board/siemens/iot2050/board.c | 39 ++++++++++++++++++++++++++---------
+ doc/board/siemens/iot2050.rst |  5 +++++
+ include/configs/iot2050.h     |  3 +++
+ 3 files changed, 37 insertions(+), 10 deletions(-)
+
+diff --git a/board/siemens/iot2050/board.c b/board/siemens/iot2050/board.c
+index b83d5f8669e8..d6228b9bd26b 100644
+--- a/board/siemens/iot2050/board.c
++++ b/board/siemens/iot2050/board.c
+@@ -38,6 +38,8 @@ struct iot2050_info {
+ 	u8 mac_addr_cnt;
+ 	u8 mac_addr[8][ARP_HLEN];
+ 	char seboot_version[40 + 1];
++	u8 padding[3];
++	u32 ddr_size_mb;
+ } __packed;
+ 
+ /*
+@@ -341,25 +343,42 @@ int board_init(void)
+ 
+ int dram_init(void)
+ {
+-	if (board_is_advanced())
+-		gd->ram_size = SZ_2G;
+-	else
+-		gd->ram_size = SZ_1G;
++	struct iot2050_info *info = IOT2050_INFO_DATA;
++	gd->ram_size = ((phys_size_t)(info->ddr_size_mb)) << 20;
+ 
+ 	return 0;
+ }
+ 
++ulong board_get_usable_ram_top(ulong total_size)
++{
++	/* Limit RAM used by U-Boot to the DDR low region */
++	if (gd->ram_top > 0x100000000)
++		return 0x100000000;
++
++	return gd->ram_top;
++}
++
+ int dram_init_banksize(void)
+ {
+ 	dram_init();
+ 
+-	/* Bank 0 declares the memory available in the DDR low region */
+-	gd->bd->bi_dram[0].start = CFG_SYS_SDRAM_BASE;
+-	gd->bd->bi_dram[0].size = gd->ram_size;
++	if (gd->ram_size > SZ_2G) {
++		/* Bank 0 declares the memory available in the DDR low region */
++		gd->bd->bi_dram[0].start = CFG_SYS_SDRAM_BASE;
++		gd->bd->bi_dram[0].size = SZ_2G;
++
++		/* Bank 1 declares the memory available in the DDR high region */
++		gd->bd->bi_dram[1].start = CFG_SYS_SDRAM_BASE1;
++		gd->bd->bi_dram[1].size = gd->ram_size - SZ_2G;
++	} else {
++		/* Bank 0 declares the memory available in the DDR low region */
++		gd->bd->bi_dram[0].start = CFG_SYS_SDRAM_BASE;
++		gd->bd->bi_dram[0].size = gd->ram_size;
+ 
+-	/* Bank 1 declares the memory available in the DDR high region */
+-	gd->bd->bi_dram[1].start = 0;
+-	gd->bd->bi_dram[1].size = 0;
++		/* Bank 1 declares the memory available in the DDR high region */
++		gd->bd->bi_dram[1].start = 0;
++		gd->bd->bi_dram[1].size = 0;
++	}
+ 
+ 	return 0;
+ }
+diff --git a/doc/board/siemens/iot2050.rst b/doc/board/siemens/iot2050.rst
+index ee3c5c958464..c6bb06750584 100644
+--- a/doc/board/siemens/iot2050.rst
++++ b/doc/board/siemens/iot2050.rst
+@@ -29,6 +29,11 @@ The following binaries from that source need to be present in the build folder:
+  - seboot_pg1.bin
+  - seboot_pg2.bin
+ 
++Starting from SE-Boot D/V01.04.0x.0x, the DDR size is passed from the SE-Boot to
++the u-boot, this is not compatible with the old version u-boot (<=2023.10)
++anymore. Make sure to use the latest u-boot version, or version contains the
++commit "board: siemens: iot2050: Pass DDR size from FSBL".
```

Comment:
"Note that SE-Boot D/V01.04.01.02 or greater is required, otherwise the DDR size
will not be picked up correctly by U-Boot."

I makes no sense to ask for a U-Boot version greater than X in the U-Boot source tree.

Also, we do not release Vxx

Reply:
This one is still pending.

Reply:
Fixed